src/wristpy/processing/calibration.py

```diff
@@ -417,6 +433,59 @@ def _closest_point_fit(self, no_motion_data: np.ndarray) -> LinearTransformation
 
         return LinearTransformation(scale=scale, offset=offset)
 
+    def _closest_point_fit_gradient_descent(
+        self, no_motion_data: np.ndarray
+    ) -> LinearTransformation:
+        """Find linear transformation parameters that minimzes distance to unit sphere.
+
+        This function implements the scipy optimize.minimze function to find the
+        optimial scale and offset parameters that will minimze the error function,
+        which is defined as the distance of the no_motion acceleration data
+        from the unit sphere (where the unit sphere represents the ideal data points
+        under 1g acceleartion due to Earth's gravity). The initial guess for the
+        scale and offset are chosen by the equivalent transformation that would z-score
+        the no_motion_data.
+
+        Args:
+            no_motion_data: The acceleration data during periods of no motion.
+
+        Returns:
+            A LinearTransformation object with scale and offset parameters to be applied
+            to acceleration data for calibration.
+
+        Raises:
+            ValueError if the optimization has failed.
+        """
+        start_scale = 1 / (np.sqrt(3) * np.std(no_motion_data, axis=0))
+        start_offset = -np.mean(no_motion_data, axis=0) * start_scale
+
+        def get_distance_to_unit_sphere(params: list[float]) -> float:
+            scale = params[:3]
+            offset = params[3:]
+            distances = np.linalg.norm((no_motion_data * scale) + offset, axis=1) - 1
+            return np.sum(distances**2)
```

Comment:
I would recommend computing the gradient yourself rather than relying on scipy numerical gradient estimation. Will be more precise and reduce function evals by ~10x.

```suggestion
            transformed = scale * no_motion_data + offset
            norms = np.linalg.norm(transformed, axis=-1)
            deltas = norms - 1.0
            obj = (deltas ** 2).mean()

            scaled_transformed = (deltas[:, None] / norms[:, None]) * transformed
            grad_offset = scaled_transformed.mean(axis=0)
            grad_scale = (scaled_transformed * no_motion_data).mean(axis=0)
            grad = np.concatenate([grad_scale, grad_offset])
            return obj, grad
```